Sources/GRPC/Docs.docc/Proposals/0001-stub-api.md

```diff
@@ -0,0 +1,1100 @@
+      print("'Get' failed with error code '\(error.code)' and metadata '\(error.metadata)'")
+    }
+  }
+
+  // Execute the RPC (sugared API):
+  await echo.get(request: request) { response in
+    print("'Get' received metadata '\(response.metadata)'")
+    do {
+      let message = try response.message
+      print("'Get' received '\(message.text)'")
+    } catch {
+      print("'Get' caught error '\(error)'")
+    }
+  }
+
+  // The generated code _could_ default the closure to return the response
+  // message which would make the common case straighforward:
+  let message = try await echo.get(request: request)
+  print("'Get' received '\(message.text)'")
+}
+
+func clientStreaming(echo: some EchoClientProtocol) async throws {
+  // Make the request:
+  let request = ClientRequest.Stream<Echo.Request> { writer in
+    for text in ["foo", "bar", "baz"] {
+      try await writer.write(Echo.Request(text: text))
+    }
+  }
+
+  // Execute the RPC:
+  try await echo.collect(request: request) { response in
+    // (Same as for the unary "Get")
+  }
+}
+
+func serverStreaming(echo: some EchoClientProtocol) async throws {
+  // Make the request, adding metadata:
+  let request = ClientRequest.Single(
+    message: Echo.Request(text: "foo bar baz"),
+    metadata: ["foo": "bar"],
+  )
+
+  // Execute the RPC (most verbose API):
+  try await echo.expand(request: request) { response in
+    switch response.result {
+    case .success(let body):
+      print("'Expand' accepted with metadata '\(body.metadata)'")
+      do {
+        for try await part in body.bodyParts {
+          switch part {
+          case .message(let message):
+            print("'Expand' received message '\(message.text)'")
+          case .trailers(let metadata):
+            print("'Expand' received trailers '\(metadata)'")
+          }
+        }
+      } catch let error as RPCError {
+        print("'Expand' failed with error '\(error.code)' and metadata '\(error.metadata)'")
+      } catch {
+        print("'Expand' failed with error '\(error)'")
+      }
+    case .failure(let error):
+      print("'Expand' rejected with error '\(error.code)' and metadata '\(error.metadata)'")
+    }
+  }
+
+  // Execute the RPC (sugared API):
+  await echo.expand(request: request) { response in
+    print("'Expand' received metadata '\(response.metadata)'")
+    do {
+      for try await message in response.messages {
+        print("'Expand' received '\(message.text)'")
+      }
+    } catch let error as RPCError {
+      print("'Expand' failed with error '\(error.code)' and metadata '\(error.metadata)'")
+    } catch {
+      print("'Expand' failed with error '\(error)'")
+    }
+  }
+
+  // Note: there is no generated 'default' handler, the function body defines
+  // the lifetime of the RPC and the RPC is cancelled once the closure exits.
+  // Therefore escaping the message sequence would result in the sequence
+  // throwing an error. It must be consumed from within the handler.
```

Comment:
In the future, we should see if can adopt `~Escapable` here and let the compiler make sure this is enforced.

Reply:
Is there a timeline for this?

Reply:
This was part of the `BufferView` proposal.

Reply:
For reference: https://github.com/atrick/swift-evolution/blob/bufferview-roadmap/visions/language-support-for-BufferView.md